packages/plugin-e2e/src/models/GrafanaPage.ts

```diff
@@ -0,0 +1,47 @@
+import { Expect, Locator } from '@playwright/test';
+import { PluginTestCtx } from '../types';
+
+/**
+ * Base class for all Grafana pages.
+ *
+ * Exposes methods for locating Grafana specific elements on the page
+ */
+export abstract class GrafanaPage {
+  constructor(public readonly ctx: PluginTestCtx, protected readonly expect: Expect<any>) {}
+
+  /**
+   * Get a locator for a Grafana element by data-testid or aria-label
+   * @param selector the data-testid or aria-label of the element
+   * @param root optional root locator to search within. If no locator is provided, the page will be used
+   */
+  getByTestIdOrAriaLabel(selector: string, root?: Locator): Locator {
+    if (selector.startsWith('data-testid')) {
+      return (root || this.ctx.page).getByTestId(selector);
+    }
+
+    return (root || this.ctx.page).locator(`[aria-label="${selector}"]`);
+  }
```

Comment:
I do see the reason for adding this but it would be nice to push the user to use either test id or aria label. But given the structure we have in grafana core I guess it would be non-trivial to get in place.

Reply:
Yes! However, lately steps have been taken to migrate all selectors from aria-label to data-testid. But even if we'd be able to get to that point reasonably soon in grafana core, this package would still need to function with much older versions of Grafana so we're gonna have to keep this abstraction for some time. =/